composer/utils/misc.py

```diff
@@ -28,6 +31,18 @@ def is_model_ddp(model: torch.nn.Module) -> bool:
     return isinstance(model, DistributedDataParallel)
 
 
+def is_model_fsdp(model: torch.nn.Module) -> bool:
+    """Whether any submodule of ``model`` is an instance of a :class:`.FullyShardedDataParallel`."""
+    if version.parse(torch.__version__) < version.parse('1.12.0'):
+        return False
+    else:
+        from torch.distributed.fsdp import FullyShardedDataParallel
+        for module in model.modules():
+            if isinstance(module, FullyShardedDataParallel):
+                return True
+        return False
```

Comment:
Unless I'm missing something, this is only used in one place (fsdp_enabled property). Can we just remove this abstraction layer and put this code directly there?

Reply:
Yeah that sounds good to me!